backend/remote/remote.go

```diff
@@ -175,12 +177,21 @@ func (r *Remote) monitorPing(client *rpc.Client) error {
 	ticker := time.NewTicker(pingInveral)
 	defer ticker.Stop()
 
+	retry := 0
 	for {
 		select {
 		case <-r.closeChan:
 			return nil
 		case <-ticker.C:
-			if err := r.Ping(); err != nil {
+			if err := r.Ping(); err == nil {
+				retry = 0 // reset on success
+				break
+			} else {
+				if err == ErrPingTimeout && retry < pingRetries {
```

Comment:
you think of retrying on the whole net.Error class? 
- some of them are not transitive 
- but then we give up after retrying anyway...
Yeah, makes sense to me